cranelift-codegen/src/verifier/locations.rs

```diff
@@ -49,11 +53,19 @@ impl<'a> LocationVerifier<'a> {
     fn check_constraints(&self, errors: &mut VerifierErrors) -> VerifierStepResult<()> {
         let dfg = &self.func.dfg;
         let mut divert = RegDiversions::new();
+        let mut entry_divert = EntryRegDiversions::new();
 
         for ebb in self.func.layout.ebbs() {
```

Comment:
More a question to myself than anything, since I'm not sure about this: Is this an adequate iteration order after splitting edges?  The edge splitter inserts new blocks near the end of the ebb list, so the layout will not be compatible with the dominator tree, say.

Reply:
the branch splitting only adds blocks right after the current one (for the terminator), or almost at the end (for conditional branches).

The regalloc will follow the reversed post-order, which should work fine for forwarding diversion to blocks which have a single predecessors.

This algorithm should not have any issues with new blocks added by branch splitting but will wrongly report errors on blocks with a single predecessors which were already present in the CFG before the branch splitting. We should change this to either the topo-order or the reversed post-order.